### ISSUE TITLE
10.6.4 UMD compat throws undefined (reading 'useState')

### ISSUE THREAD

**Issue by kavika-1** [preactjs/preact#3434]
- [ ] Check if updating to the latest Preact version resolves the issue

Appears issue starts with current latest, `10.6.4`.  Works fine with `10.6.3`.

**Describe the bug**
Loading preact > hooks > compat from unpkg (and other cdns) `10.6.3` works fine.  Loading latest `10.6.4` or @10 throws the following error:
```
index.js:152 Uncaught TypeError: Cannot read properties of undefined (reading 'useState')
    at index.js:152:2
    at compat.umd.js:1:206
    at compat.umd.js:1:244
```
Maybe a recent build process change introduced a UMD wrapper issue?

**To Reproduce**

1. Include umd preact, hooks, compat as script tags in `<head>` of an html page.
2. Load the page in chrome

Result is the above error.  Also, window.preact looks correct.  preactHooks looks ok.  preactCompat is `{}`.

**Expected behavior**
No errors in the console, and window.preactCompat contains all methods/properties necessary to alias to React/ReactDOM.  That is the case with `10.6.3`.

** Repro html ***
```
<!DOCTYPE html>
<html lang="en">

<head>

  <!-- Working -->
  <!-- <script src="https://unpkg.com/preact@10.6.3/dist/preact.umd.js"></script>
  <script src="https://unpkg.com/preact@10.6.3/hooks/dist/hooks.umd.js"></script>
  <script src="https://unpkg.com/preact@10.6.3/compat/dist/compat.umd.js"></script> -->

  <!-- Broken -->
  <script src="https://unpkg.com/preact@10.6.4/dist/preact.umd.js"></script>
  <script src="https://unpkg.com/preact@10.6.4/hooks/dist/hooks.umd.js"></script>
  <script src="https://unpkg.com/preact@10.6.4/compat/dist/compat.umd.js"></script>

</head>

<body>

  <script>
    console.log('preact', preact)
    console.log('preactHooks', preactHooks)
    console.log('preactCompat', preactCompat)
  </script>

</body>

</html>
```

**Comment by JoviDeCroock** [comment]
Quickly tried it out and it seems to work 😅 https://codesandbox.io/s/musing-carson-ey3iy?file=/index.html

**Comment by JoviDeCroock** [comment]
Hey,

This seems fixed in the just published version of 10.6.5 https://codesandbox.io/s/musing-carson-ey3iy?file=/index.html going to close this issue, feel free to reopen if you run into more weirdness

**Comment by kavika-1** [comment]
Confirmed - thank you!
